### PR TITLE
feat: include workspace name and braces around branch in window title (#56)

### LINKED ISSUE #56: Feature: Title bar - orange branch in curly braces, add workspace name
## Feature Request

Improve the title bar to better distinguish the branch name and display the workspace name.

### Proposed changes

1. **Branch display** - Wrap the branch name in curly braces (e.g., `{my-feature-branch}`) and color it orange to visually isolate it from surrounding text.
2. **Workspace name** - Add the workspace name to the title bar, styled in the same color as the "Arborist" title text.

### Example

`Arborist - my-workspace {my-feature-branch}`

Where "Arborist" and "my-workspace" share the same color, and `{my-feature-branch}` is orange.

### PR DESCRIPTION
**Pull request by mcaden**
Closes #56.

Format change for the OS-native main window title:

| Build       | Workspace bound | Title                              |
| ----------- | --------------- | ---------------------------------- |
| canonical   | no              | `Arborist`                       |
| canonical   | yes             | `Arborist - <workspace>`         |
| feature     | no              | `Arborist {<branch>}`            |
| feature     | yes             | `Arborist - <workspace> {<branch>}` |

The curly braces visually isolate the build branch from surrounding text, and the workspace basename makes it clear which workspace the window belongs to when multiple Arborist instances are open in parallel.

### Scope note on color

The issue describes coloring the branch orange and the workspace name in the same color as `Arborist`. The current title bar is the OS-native window title bar, which does not support colored text. Per the implementation discussion, this PR updates the title text only; the colored styling is left for a future custom in-app title bar.

### Implementation

- `window_title_for_branch` is replaced by `window_title(branch, workspace_root: Option<&Path>)`. The canonical-build collapse now reuses `store_layout::is_canonical_build` so the title-bar story and the storage-scoping story stay coupled to one rule.
- `workspace_basename` returns `Path::file_name` lossily stringified; filesystem roots return `None` so the workspace segment is omitted rather than printed empty.
- The title is set three times across the binding lifetime:
  1. Start of `run::setup` with `workspace=None` (covers the brief startup window before workspace bind).
  2. After `boot_select_workspace` succeeds, with `Some(&binding.workspace_root)`.
  3. After every successful `workspace_switch`. The wrapper reads the currently-bound workspace from `ctx.workspace` rather than the returned `WorkspaceSwitchResult.workspace_root` so out-of-order resumption of queued switches still leaves the title matching the runtime's actual binding.

### Tests

11 new pure-fn unit tests cover:

- The full 2x2 matrix of (canonical/feature build x bound/unbound workspace).
- Whitespace trimming in the branch component.
- `workspace_basename` edge cases: trailing separator, filesystem root (Unix and Windows gated), Unicode preservation.

Verification: `cargo fmt`, `cargo clippy --workspace --all-targets -- -D warnings`, `cargo test --workspace`, `npm run lint`, `npm test -- --run` all green locally.

> 🤖 This pull request was prepared by Copilot CLI.

### CODE REVIEW

File: src-tauri/src/commands/session.rs [hunk 1]
```diff
@@ -1508,7 +1508,20 @@ pub async fn workspace_switch_impl(
...
+    // Refresh the OS window title so the new workspace name is visible (issue #56). Read the *currently bound* workspace from `ctx.workspace` rather
+    // than `result.workspace_root`, so if two switches race and resume out of order after the inner barrier releases, the title still ends up
+    // matching the workspace the runtime is actually bound to. Best-effort: failures only affect the title, not the switch result.
+    if let Some(window) = app_handle.get_webview_window("main") {
+        let current_root = ctx.workspace.read().expect("workspace lock poisoned").workspace_root.clone();
+        let title = crate::window_title(crate::BUILD_BRANCH, current_root.as_deref());
+        if let Err(err) = window.set_title(&title) {
+            tracing::warn!(%err, "failed to update main window title after workspace switch");
+        }
+    }
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src-tauri/src/commands/session.rs:1533]: 🤖 AI agent reply (acting for @mcaden):

Done in 9b054cc. The wrapper now holds the `ctx.workspace` read guard across the `set_title` call rather than read-clone-drop-then-set. The inner's commit path takes `ctx.workspace.write()`, which cannot acquire while any reader is active, so no newer scope can be committed between the `window_title` computation and the OS title-set — a delayed wrapper therefore can never overwrite a newer correct title with a stale value, even across 3+ rapid back-to-back switches.

Considered the generation-token approach you suggested, but a check-token-then-set_title sequence still has TOCTOU between the check and the OS call (no atomic compare-and-set wraps the platform API). The read-guard approach is strictly correct under `RwLock` semantics with no extra state, and the cost is just one synchronous `SetWindowTextW` / `NSWindow.title=` / `gtk_window_set_title` call held under the read lock — none re-enter into Arborist workspace code, so no deadlock surface.

Note (in case it matters for your mental model): the title can still be transiently stale between an inner commit and that wrapper's own `set_title` running. The stronger accurate claim is *no wrapper can apply a stale title after a newer scope has committed*.

File: src-tauri/src/commands/session.rs [hunk 2]
```diff
@@ -1508,7 +1508,20 @@ pub async fn workspace_switch_impl(
...
         .app_data_dir()
         .map_err(|e| AppError::new("Io", format!("app_data_dir: {e}")))?;
-    workspace_switch_impl_inner(ctx, &app_data_dir, crate::BUILD_BRANCH, new_path).await
+    let result = workspace_switch_impl_inner(ctx, &app_data_dir, crate::BUILD_BRANCH, new_path).await?;
+
+    // Refresh the OS window title so the new workspace name is visible (issue #56). Read the *currently bound* workspace from `ctx.workspace` rather
+    // than `result.workspace_root`, so if two switches race and resume out of order after the inner barrier releases, the title still ends up
+    // matching the workspace the runtime is actually bound to. Best-effort: failures only affect the title, not the switch result.
+    if let Some(window) = app_handle.get_webview_window("main") {
+        let current_root = ctx.workspace.read().expect("workspace lock poisoned").workspace_root.clone();
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src-tauri/src/commands/session.rs]: 🤖 AI agent reply (acting for @mcaden):

Done in 9b054cc. Replaced the `.expect("workspace lock poisoned")` with a `match` that logs-and-skips on poison rather than crashing — the title refresh is best-effort and shouldn't take down the process. Matches the existing pattern in `commands/mod.rs:302,371` where the PtySink status/ai-session callbacks similarly log-and-skip on workspace-lock poison instead of unwinding.
